RecoLocalCalo/HGCalRecProducers/plugins/HGCalCLUEAlgo.cc

```diff
@@ -271,8 +271,9 @@ void HGCalCLUEAlgo::calculateDistanceToHigher(const HGCalLayerTiles& lt, const u
           unsigned int otherId = lt[binId][j];
 
           float dist = distance(i, otherId, layerId);
-          bool foundHigher = cellsOnLayer.rho[otherId] > cellsOnLayer.rho[i];
-
+          bool foundHigher =
+              (cellsOnLayer.rho[otherId] > cellsOnLayer.rho[i]) ||
+              (cellsOnLayer.rho[otherId] == cellsOnLayer.rho[i] && cellsOnLayer.detid[otherId] > cellsOnLayer.detid[i]);
```

Comment:
Is the case with equal rho supposed to be ordered by detid or is this an attempt to introduce a predictable order and a random choice is a more appropriate choice?
If a random choice is more preferable, perhaps some better effort to randomize the choice in a predictable way should be made.

Reply:
@slava77 the order does not matter as soon as only one seed is created instead of 2 or 3. The effect is that these clusters will be merged in a single one.